### PR TITLE
CV2-4233 add retry looper for alegre requests that intermittently fail

### PR DESCRIPTION
**Pull request by DGaffney**
## Description

When running via smooch, every once in a while, a request from Check-API to Alegre fails. There is every indication that this is some sort of intermittent, non-serious issue. When trying to reproduce them, even at large scales with lots of attempts to recreate the issue, Alegre responds correctly consistently. When looking at Alegre and Presto, neither show any indication that there is an issue. The response itself doesn't seem to be loud or fundamentally broken. For the time being, we're introducing a retry - and if that fails, we'll be much more confident that something more fundamental is happening that requires a deeper dive.

References: CV2-4233

## How has this been tested?

I've added a direct test that should yield a proper null output after retrying multiple times, and ran a local script with side-effecting printouts to ensure it is doing what is intended.

## Things to pay attention to during code review

Nothing really - fairly straightforward retry cycle

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

**Review by melsawy — APPROVED** [review]

File: test/models/bot/alegre_v2_test.rb
Comment (caiosba) [inline test/models/bot/alegre_v2_test.rb]: Request: Fix this because it should be the opposite: `assert_equal expected, actual`

File: test/models/bot/alegre_v2_test.rb
Comment (caiosba) [inline test/models/bot/alegre_v2_test.rb:289]: Suggestion: I would simplify this because for this test the payload itself doesn't matter... something like:
```
WebMock.stub_request(:post, "#{CheckConfig.get('alegre_host')}/similarity/sync/audio").to_return(body: '{}')
```

File: app/models/concerns/alegre_v2.rb
Comment (caiosba) [inline app/models/concerns/alegre_v2.rb:237]: Suggestion for better readability: Use `nil?` instead of ` == nil`, like this:
```
while (response.nil? || response["result"].nil?) && retries < 3
```

**Review by caiosba — CHANGES_REQUESTED** [review]
Good to go once tests pass.

**Comment by DGaffney** [comment]
@caiosba ran into some slightly weird speedbumps yesterday but back to full passage - lemme know if you need anything else before signoff!

**Pull request merged**
